### PR TITLE
Made the letter FSA. TOTALLY DOPE.

### PR DESCRIPTION
**Pull request by seanrogers2657**
Made the letter FSA in scanner.cs. Updated ERROR and LETTER in constants.cs. 

img_1221

### CODE REVIEW

**Comment by JesseStephenBrown** [comment]
Hey bro, did you forget about reserved words?

**Comment by brianpmaher** [comment]
Was about to +1 this; forgot about reserved words too -.-

File: constants.cs [hunk 1]
```diff
@@ -77,9 +77,14 @@ public static class Constants {
     // Error messages
     public const string ERROR_FILE_FORMAT = "ERROR: File format does not conform to format: <filename>.mp; ";
     public const string ERROR_DISPATCHER_DIGIT = "ERROR: Dispatcher done goofed. Passed {0} to the digit FSA, but {0} is not a digit.";
+    public const string ERROR_DISPATCHER_LETTERS = "ERROR: Dispatcher failed miserably. Passed {0} to the letter FSA, but {0} is not a valid identifier.";
```
Comment (brianpmaher) [inline constants.cs]: This goes over the 100 character limit, but we can deal with that in the refactor ticket.

File: constants.cs [hunk 2]
```diff
@@ -77,9 +77,14 @@ public static class Constants {
...
 
     // List of whitespace characters
     public const string WHITESPACE = " \r\n\t";
 
     public const string DIGITS = "0123456789";
+
+    // List of the letters
+    public const string LETTERS = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
```
Comment (brianpmaher) [inline constants.cs]: Going to conflict with Jesse's PR, but doesn't matter.. Just sayin

**Comment by brianpmaher** [comment]
Take a look at this. http://www.dotnetperls.com/string-switch
While it's incredibly ugly, you should just do a big switch case with the possible strings for all reserved words in the accept state within the else clause (an ID has been found) and the default case in the switch statement should just return a generic identifier type (given that it's like a variable or something)

**Comment by brianpmaher** [comment]
No need to close the pull request, just next time you do a add/commit/push, it will show up here

**Pull request closed without merging**